document-store/src/main/java/org/hypertrace/core/documentstore/query/GenericOrderBy.java

```diff
@@ -0,0 +1,14 @@
+package org.hypertrace.core.documentstore.query;
+
+import lombok.Value;
+import org.hypertrace.core.documentstore.expression.Sortable;
+
+/**
+ * A generic ORDER BY definition that supports expressions. Note that this class is a more general
+ * version of {@link org.hypertrace.core.documentstore.OrderBy}
+ */
+@Value(staticConstructor = "of")
+public class GenericOrderBy {
```

Comment:
The use of the term "Generic" here seems strange - isn't everything in this tree generic? If the idea is to keep it distinct from that other class, I think the package name would be the appropriate place to make that distinction (and a corresponding comment similar to what's here).